lib/ansible/galaxy/collection.py

```diff
@@ -401,9 +404,10 @@ def from_name(collection, apis, requirement, force, parent=None):
                 else:
                     resp = api.get_collection_versions(namespace, name)
 
-                    # Galaxy supports semver but ansible-galaxy does not. We ignore any versions that don't match
-                    # StrictVersion (x.y.z) and only support pre-releases if an explicit version was set (done above).
-                    versions = [v for v in resp if StrictVersion.version_re.match(v)]
+                    if allow_pre_release:
+                        versions = resp
+                    else:
+                        versions = [v for v in resp if not SemanticVersion(v).is_prerelease]
```

Comment:
If the version is unspecified and latest release is a prerelease you get an error like 
```
ERROR! Cannot meet requirement * for dependency namespace.name from source 'https://galaxy.ansible.com/api/'. Available versions before last requirement added: 
Requirements from:
	base - 'namespace.name:*'
```
It might be nice if there was a hint to use the new flag if only prereleases are found and the user isn't trying to install a particular version.

Reply:
I'll have to take a look.  I think I'll have to extend `CollectionRequirement` to accept a list of pre-releases, or move the filtering into `CollectionRequirement` itself.  As I say it, I'm thinking the later may be best.

Reply:
Updated! Please let me know what you think:

```
ERROR! Cannot meet requirement * for dependency freeipa.ansible_freeipa from source 'https://galaxy.ansible.com/api/'. Available versions before last requirement added:
Requirements from:
	base - 'freeipa.ansible_freeipa:*'
This collection only contains pre-releases. Utilize `--pre` to install pre-releases, or explicitly provide the pre-release version.
```